shell/src/main/java/alluxio/cli/validation/HdfsImpersonationValidationTask.java

```diff
@@ -0,0 +1,264 @@
+                + "in Alluxio but not in HDFS.", userName, missedGroups));
+        advice.append(String.format("Please add the missing groups to %s. ", hdfsKey));
+        continue;
+      }
+
+      // All checks passed
+      msg.append(String.format("Found matching configuration in %s and %s. ",
+              alluxioKey.toString(), hdfsKey));
+    }
+    return new ValidateUtils.TaskResult(state, taskName, msg.toString(), advice.toString());
+  }
+
+  // At least the current host should be in hadoop.proxy.<username>.hosts
+  private ValidateUtils.TaskResult validateImpersonationHosts() {
+    String taskName = "Validate proxyuser hosts";
+    ValidateUtils.State state = ValidateUtils.State.OK;
+    StringBuilder msg = new StringBuilder();
+    StringBuilder advice = new StringBuilder();
+
+    // current host
+    String localhost = NetworkAddressUtils.getLocalHostName(1000);
+
+    // All the possible users Alluxio needs
+    Set<String> allUsers = Sets.union(mImpersonationUsers.keySet(), mImpersonationGroups.keySet());
+    for (String userName : allUsers) {
+      msg.append(String.format("User %s is configured to allow impersonation is Alluxio. ",
+              userName));
+
+      String hdfsKey = String.format("hadoop.proxyuser.%s.hosts", userName);
+      if (!mCoreConf.containsKey(hdfsKey)) {
+        state = ValidateUtils.State.FAILED;
+        msg.append("But the user is not allowed to use impersonation on this host.");
+        advice.append(String.format("Please configure %s to contain %s. ", hdfsKey, localhost));
+        continue;
+      }
+
+      String proxyHosts = mCoreConf.get(hdfsKey);
+      // Consider wildcard separately
+      if (proxyHosts.equals(ImpersonationAuthenticator.WILDCARD)) {
+        msg.append(String.format("The user is enabled for impersonation from all hosts. "));
+        continue;
+      }
+      // If wildcard is not used and the localhost is not in the permitted list
+      if (!proxyHosts.contains(localhost)) {
+        state = ValidateUtils.State.FAILED;
+        msg.append(String.format("But %s does not contain host %s. ", hdfsKey, localhost));
+        advice.append(String.format("Please enable host %s in %s. ", localhost, hdfsKey));
+        continue;
+      }
+
+      // Passed all checks
+      msg.append(String.format("Host %s is enabled to use impersonation in HDFS. ", localhost));
+    }
+    return new ValidateUtils.TaskResult(state, taskName, msg.toString(), advice.toString());
+  }
+
+  @Override
+  public ValidateUtils.TaskResult validate(Map<String, String> optionMap) {
+    if (shouldSkip()) {
+      return new ValidateUtils.TaskResult(ValidateUtils.State.SKIPPED, getName(),
+              mMsg.toString(), mAdvice.toString());
+    }
+
+    ValidateUtils.TaskResult loadConfig = loadHdfsConfig();
+    if (loadConfig.getState() != ValidateUtils.State.OK) {
+      return loadConfig;
+    }
+
+    // TODO(jiacheng): do we want to check the current user even if
```

Comment:
Do we want to assume the current running user needs impersonation? I don't think so.